plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -404,20 +423,24 @@ public void onOutputLine(String line) {
       } else if (line.contains("Error: A fatal exception has occurred. Program will exit")) { //$NON-NLS-1$
         // terminate the Python process
         stop(false);
-      } else if ((matcher = moduleStartedPattern.matcher(line)).matches()) {
+      } else if ((matcher = moduleStartedPattern.matcher(line)).matches()
+          || (matcher = moduleRunningPattern.matcher(line)).matches()) {
         String serviceId = matcher.group("service");
-        moduleToUrlMap.put(serviceId, matcher.group("url"));
-        int port = parseInt(matcher.group("port"), 0);
+        String url = matcher.group("url");
+        moduleToUrlMap.put(serviceId, url);
+        String portString = matcher.group("port");
+        int port = parseInt(portString, 0);
         if (port > 0 && (serverPortCandidate == 0 || "default".equals(serviceId))) { // $NON-NLS-1$
           serverPortCandidate = port;
         }
-      } else if ((matcher = adminStartedPattern.matcher(line)).matches()) {
+      } else if ((matcher = adminStartedPattern.matcher(line)).matches()
+          || (matcher = adminRunningPattern.matcher(line)).matches()) {
         int port = parseInt(matcher.group("port"), 0);
-        if (port > 0 && adminPort == 0) {
+        if (port > 0 && adminPort <= 0) {
```

Comment:
I think there was a reason we used `==` here, which I can't remember exactly. Maybe it was a conceptual distinction: -1 meaning never initialized, and 0 meaning we let the server pick a port. Maybe not a big deal, so I'm OK as long as it's working.